### ISSUE TITLE
No schema type found for dotnet type Char. Make sure you add it or add a type mapping

### ISSUE THREAD

**Issue by bzbetty** [EntityGraphQL/EntityGraphQL#182]
Upon upgrading to 3.0 I get this error while deep in EQL before I can manually add the type.

```
   at EntityGraphQL.Schema.SchemaProvider`1.GetSchemaType(Type dotnetType, QueryRequestContext requestContext)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.CacheType(Type propType, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.<ProcessFieldOrProperty>d__6.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at EntityGraphQL.Schema.SchemaBuilder.GetFieldsFromObject(Type type, ISchemaProvider schema, SchemaBuilderOptions options, Boolean isInputType)
   at EntityGraphQL.Schema.SchemaBuilder.FromObject[TContextType](SchemaProvider`1 schema, SchemaBuilderOptions options)
   at EntityGraphQL.Schema.SchemaProvider`1.PopulateFromContext(SchemaBuilderOptions options)
   at EntityGraphQL.AspNet.EntityGraphQLAspNetServiceCollectionExtensions.AddGraphQLSchema[TSchemaContext](IServiceCollection serviceCollection, Action`1 configure)
   at EntityGraphQL.AspNet.EntityGraphQLAspNetServiceCollectionExtensions.AddGraphQLSchema[TSchemaContext](IServiceCollection serviceCollection, Action`1 configure)
   at StartUp.Configure(IFunctionsHostBuilder builder) 
```

**Comment by bzbetty** [comment]
Worked around for now using PreBuildSchemaFromContext to register types.

**Comment by lukemurray** [comment]
Do you know what type it was? And where in the object graph it was? Would like to figure out why it was not automatically added.

**Comment by bzbetty** [comment]
I have a class with a 'char' property on it, doesn't seem to be anything particularly special about it.

I'll see if I can make a replication test

**Comment by lukemurray** [comment]
Fixed in 3.0.3
